test/Microsoft.TestPlatform.Utilities.UnitTests/CodeCoverageRunSettingsProcessorTests.cs

```diff
@@ -0,0 +1,308 @@
+            @"        <Attribute>^System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute$</Attribute>",
+            @"        <Attribute>^Microsoft.VisualStudio.TestPlatform.TestSDKAutoGeneratedCode.*</Attribute>",
+            @"      </Exclude>",
+            @"    </Attributes>",
+            @"    <Sources>",
+            @"      <Exclude>",
+            @"        <Source>.*\\atlmfc\\.*</Source>",
+            @"        <Source>.*\\vctools\\.*</Source>",
+            @"        <Source>.*\\public\\sdk\\.*</Source>",
+            @"        <Source>.*\\externalapis\\.*</Source>",
+            @"        <Source>.*\\microsoft sdks\\.*</Source>",
+            @"        <Source>.*\\vc\\include\\.*</Source>",
+            @"        <Source>.*\\msclr\\.*</Source>",
+            @"        <Source>.*\\ucrt\\.*</Source>",
+            @"      </Exclude>",
+            @"    </Sources>",
+            @"    <CompanyNames/>",
+            @"    <PublicKeyTokens/>",
+            @"  </CodeCoverage>",
+            @"</Configuration>");
+
+        private CodeCoverageRunSettingsProcessor processor;
+
+        private XmlDocument document;
+
+        private XmlNode ExtractNode(XmlNode node, string path)
+        {
+            try
+            {
+                return node.SelectSingleNode(path);
+            }
+            catch
+            {
+            }
+
+            return null;
+        }
+
+        private Tuple<XmlNode, XmlNode> ExtractNodes(XmlNode currentSettingsRoot, XmlNode defaultSettingsRoot, string path)
+        {
+            var currentNode = this.ExtractNode(currentSettingsRoot, path);
+            var defaultNode = this.ExtractNode(defaultSettingsRoot, path);
+            Assert.IsNotNull(currentNode);
+            Assert.IsNotNull(defaultNode);
+
+            return new Tuple<XmlNode, XmlNode>(currentNode, defaultNode);
+        }
+
+        private void CompareResults(XmlNode currentSettingsRoot, XmlNode defaultSettingsRoot, string path)
+        {
+            var nodes = this.ExtractNodes(currentSettingsRoot, defaultSettingsRoot, path);
+
+            Assert.AreEqual(nodes.Item1.ChildNodes.Count, nodes.Item2.ChildNodes.Count);
+
+            var set = new HashSet<string>();
+            foreach (XmlNode child in nodes.Item1.ChildNodes)
+            {
+                if (!set.Contains(child.OuterXml))
+                {
+                    set.Add(child.OuterXml);
+                }
+            }
+
+            foreach (XmlNode child in nodes.Item2.ChildNodes)
+            {
+                if (!set.Contains(child.OuterXml))
+                {
+                    set.Add(child.OuterXml);
+                    continue;
+                }
+
+                set.Remove(child.OuterXml);
+            }
+
+            Assert.AreEqual(set.Count, 0);
+        }
+
+        public CodeCoverageRunSettingsProcessorTests()
+        {
+            this.processor = new CodeCoverageRunSettingsProcessor();
```

Comment:
remove constructor and use SetUp method so that every test starts with clean workspace

Reply:
I think there's no need to do this anymore. After the code cleanup there's virtually no persistent state created between runs. The processor class only loads the default settings document in its constructor whereas the tests' constructor loads the default settings and creates the shared processor.